src/vm/corhost.cpp

```diff
@@ -1324,6 +1325,10 @@ HRESULT CorHost2::ExecuteAssembly(DWORD dwAppDomainId,
 
     Assembly *pAssembly = AssemblySpec::LoadAssembly(pwzAssemblyPath);
 
+#if defined(FEATURE_MULTICOREJIT)
```

Comment:
I'm wondering if we need to make sure that the current process is not an AppX process before calling AutoStartProfile.  Otherwise, we might call AutoStartProfile and AutoStartAppXProfile in quick succession for AppX processes.  See https://github.com/dotnet/coreclr/pull/4616/files#diff-91a4dc7e15e8a7477233a5f660d2a0c4R8759 for the AppX call.  In your tests, do we get here, but not call AutoStartProfileAppx because it's not an AppX profile?  If so, we should do the check before calling AutoStartProfile.